addOns/spider/src/test/java/org/zaproxy/addon/spider/SvgHrefSpiderUnitTest.java

```diff
@@ -0,0 +1,212 @@
+        // Given
+        HttpMessage msg = createMessage("test.svg");
+        msg.setResponseBody(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n"
+                        + "<svg width=\"5cm\" height=\"3cm\" viewBox=\"0 0 5 3\" version=\"1.1\" xmlns=\"http://www.w3.org/2000/svg\">\n"
+                        + "  <rect x=\".01\" y=\".01\" width=\"4.98\" height=\"2.98\" fill=\"none\" stroke=\"blue\" stroke-width=\".03\"/>\n"
+                        + "</svg>");
+        // When
+        boolean parse = shs.parseResource(msg, new Source(msg.getResponseBody().toString()), 0);
+        // Then
+        assertFalse(parse);
+    }
+
+    @Test
+    void shouldNotParseResourceWhenSAXParseExceptionEncountered() {
+        // Given
+        HttpMessage msg = createMessage("test.svg");
+        msg.setResponseBody(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n"
+                        + "<svg width=\"5cm\" height=\"3cm\" viewBox=\"0 0 5 3\" version=\"1.1\" xmlns=\"http://www.w3.org/2000/svg\">\n"
+                        + "  <rect x=\".01\" y=\".01\" width=\"4.98\" height=\"2.98\" fill=\"none\" stroke=\"blue\" stroke-width=\".03\"/>\n"
+                        // The following line produces a SAXParseException other than the DOCTYPE
+                        // issue tested elsewhere due to the ampersand outside of a CDATA block
+                        + "<text x=\"20\" y=\"35\" class=\"small\">Test & Text</text>"
+                        + "</svg>");
+        // When
+        boolean parse = shs.parseResource(msg, new Source(msg.getResponseBody().toString()), 0);
+        // Then
+        assertFalse(parse);
+    }
+
+    @Test
+    void shouldNotParseResourceWithDoctypeDeclaration() {
+        // Given
+        HttpMessage msg = createMessage("test.svg");
+        msg.setResponseBody(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n"
+                        + "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\" \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n"
+                        + "<svg width=\"5cm\" height=\"3cm\" viewBox=\"0 0 5 3\" version=\"1.1\" xmlns=\"http://www.w3.org/2000/svg\">\n"
+                        + "  <rect x=\".01\" y=\".01\" width=\"4.98\" height=\"2.98\" fill=\"none\" stroke=\"blue\" stroke-width=\".03\"/>\n"
+                        + "  <a HREF=\"http://www.w3.org\">\n"
+                        + "    <ellipse cx=\"2.5\" cy=\"1.5\" rx=\"2\" ry=\"1\" fill=\"red\"/>\n"
+                        + "  </a>"
+                        + "</svg>");
+        // When
+        boolean parse = shs.parseResource(msg, new Source(msg.getResponseBody().toString()), 0);
+        // Then
+        assertFalse(parse);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"href", "HREF", "xlink:href", "XLINK:HREF"})
+    void shouldParseValidResourceWithHref(String elementName) {
```

Comment:
It would be good to also verify the value of the URL(s) found.

Reply:
I agree, however wouldn't I then have to store them locally which should probably only be done for tesitng due to memory/performance?

Reply:
Leave as is.

Reply:
Should I have added a TODO? Or should some sort of note be added to the existing spider extension issue?

Reply:
Fine either way.